torch/csrc/autograd/python_variable.cpp

```diff
@@ -105,7 +105,7 @@ static int THPVariable_traverse(THPVariable *self, visitproc visit, void *arg)
 static int THPVariable_clear(THPVariable *self)
 {
   Py_CLEAR(self->backward_hooks);
-  if (self->cdata.defined()) {
+  if (self->cdata.defined() && self->cdata.get_autograd_meta()) {
```

Comment:
Briefly discussed in person, seems like this should not be happening because semantic changes aren't being made in this PR (so variables should always have autograd_meta).